FLOE-324: Implement high-contrast themes

acheetham opened the pull request:
There are still some minor visual issues at small sizes (which are also there in the default theme), but I suggest we get this in if it's "good enough" and file JIRAs for remaining issues.

murilocrivellaric commented:
I checked the high-contrast visuals and they look good to me.

jobara commented:
I believe that all of the text elements in buttons, should have their `background-color: transparent important!` set. This should remove the flashing that happens on hover and with the visibility of the diagonal lines for disabled buttons.

![screen shot 2015-05-19 at 9 11 24 am](https://cloud.githubusercontent.com/assets/556024/7703653/66e6e22a-fe07-11e4-9e32-b6f862ff9132.png)

Also, we should remove the hover styles for disabled items.

![screen shot 2015-05-19 at 9 11 40 am](https://cloud.githubusercontent.com/assets/556024/7703661/74efbc52-fe07-11e4-9bfd-67e4bae6e0bb.png)

The "smaller" button doesn't appear to have disabled styling at the min size.

jobara commented:
Also, we should remove the hover styles for disabled items.

![screen shot 2015-05-19 at 9 11 40 am](https://cloud.githubusercontent.com/assets/556024/7703661/74efbc52-fe07-11e4-9bfd-67e4bae6e0bb.png)

jobara commented:
The "smaller" button doesn't appear to have disabled styling at the min size.

![screen shot 2015-05-19 at 9 16 24 am](https://cloud.githubusercontent.com/assets/556024/7703704/bd284dea-fe07-11e4-9683-93d8f64b003b.png)

acheetham commented:
@jobara, regarding disabled +/- buttons, I don't have any of those yet, they must only be in your branch?

acheetham commented:
@jobara, the minus button at the minimum has no disabled styling because it hasn't actually been disabled. If "disabled" is added to the element, the styling is there, so that's a different bug

jobara commented:
@acheetham could you file a jira about the smaller button not being disabled at the minimum value?

jobara commented:
@acheetham the tooltips for the language options don't have a border like all the other tooltips.
![screen shot 2015-05-20 at 11 15 30 am](https://cloud.githubusercontent.com/assets/556024/7729229/adc6d6a2-fee1-11e4-98ed-b9a1c5bfa679.png)

jobara commented:
@acheetham filed a jira about the button not disabling.
http://issues.fluidproject.org/browse/FLOE-346

jobara commented on src/css/style.css:
```diff
@@ -9,12 +9,19 @@
 body {
     background-color: #dddddd;
 }
+.row .row { /* override foundation, which causes problems in high-contrast */
```
Should this be scoped within .gpii-fd as well, like the button override.

acheetham commented:
@jobara, I've addressed your comment about scoping the row styles, but I've discovered that if I similarly scope other styles (such as the button style in line 34) it messes up other things that now become less specific than it. Fixing that would require scoping just about everything, which would make things less performant.

jobara commented:
merged at 4e9ea40684df5ccb59d9d173569bb5d0acd5ac3e
